app/jobs/hyku_addons/reindex_model_job.rb

```diff
@@ -51,9 +46,37 @@ def reindex_works(works)
       def mint_doi(work)
         return if work.doi.present?
 
+        Rails.logger.debug "=== about to mint doi for #{work.title} ==== "
         work.update(doi_status_when_public: "findable")
         register_doi = Hyrax::DOI::DataCiteRegistrar.new.register!(object: work)
         work.update(doi: [register_doi.identifier])
       end
+
+      def fetch_work_using_ids(klass)
+        @use_work_ids.map do |id|
+          work = klass.find(id)
+          Rails.logger.debug "=== updating index with #{id}for #{work&.title&.to_a&.first} ===="
+          mint_doi(work) if @cname_doi_mint.present? && @cname_doi_mint.include?(@cname)
+          work.save
+        end
+      end
+
+      # When the offset becomes too large, no records would be found
+      def reindex_and_mint_work(work_class, offset)
+        works = work_class.where("title_tesim:*").limit(@limit).offset(offset)
+        return unless works&.to_a&.any?
+
+        Rails.logger.debug "=== Starting to reindex #{@klass} in #{@cname} ==="
+
+        # with calling to_a it always returns true, even when no records found
+        reindex_works(works)
+
+        new_page_count = @page.to_i + 1
+
+        # Re-enqueue
+        ReindexModelJob.perform_later(@klass, @cname, @limit, page: new_page_count, cname_doi_mint: @cname_doi_mint)
```

Comment:
You dont need to add  new variable for the updated page count, the code is more readable without it: 

```ReindexModelJob.perform_later(@klass, @cname, @limit, page: @page.to_i + 1, cname_doi_mint: @cname_doi_mint)```

Reply:
I can make that change though i don't find your change easier to process if glancing quickly through the code compared to the current code.

Reply:
I am making the change but @prdanelli going with your desk analogy, a method with 2 to 3 local variables is like a desk with 2 or 3 books on it being called cluttered. Sounds more like a preference of having nothing on the desk and not a case of cluttering.